### DISCUSSION TITLE
Upgrade path

### DISCUSSION THREAD

**Discussion by tumtumsback** [netbox-community/netbox#7326] [Q&A]
Hello all,

I inherited an Ubuntu 16.04.6 LTS server running Netbox v2.3.1

Super far behind, I know -- what are my best options to get current?

Thanks!

**Comment by domainzero** [comment]
Hey @tumtumsback

[This documentation](https://netbox.readthedocs.io/en/stable/installation/upgrading/) covers pretty much all of it. You'll need modern Python, PostgreSQL, and Redis installed, and you'll need to update to Netbox 2.10, then Netbox 2.11 before you get to Netbox 3.0.2 according to the [3.0 release notes](https://netbox.readthedocs.io/en/stable/release-notes/).

**Comment by jeremystretch** [comment]
Since you're starting with such an old base operating system, I would strongly recommend provisioning a new system first, and installing NetBox on it. Obviously, a ton has changed since the days of v2.3, so you'll want to spend some time reading through NetBox's release notes (though it's unlikely you have much in the way of integrations).

Once you have the new system in place, you can install NetBox per the [installation docs](https://netbox.readthedocs.io/en/stable/installation/). However, you'll want to start with a v2.11 release. This will allow you to copy over the PostgreSQL database and upgrade to v2.11 automatically. Once that's done, you can proceed with upgrading from v2.11 to v3.0.

**Comment by tumtumsback** [comment]
Thank you very much! I will surely give this a try on Monday.

**Comment by tumtumsback** [comment]
Stood up a new Ubuntu 20.04.3 LTS box. Copied database from old to new box. Installed v2.11.11 successfully.

When I try to upgrade from 2.11.11 to 3.0, I get errors. I then simply just tried to do a 2.11.11 to 2.11.12 upgrade, and that keeps giving me a 503. Any help would be greatly appreciated. Thanks!

**Reply by WSU-HoneyBadger** [reply]
I had troubles with 2 things when going from 2.11.11 to 3.0

had to removing debugging for LDAP
Had to change the user for the httpd systemd startup, there is a note for this somewhere, release notes or upgrade notes.

**Reply by WSU-HoneyBadger** [reply]
It was in the "Migrating to systemd" 

https://netbox.readthedocs.io/en/stable/installation/migrating-to-systemd/

_"You may need to modify the user that the systemd service runs as. Please verify the user for httpd on your specific release and edit both files to match your httpd service under user and group. The username could be "nobody", "nginx", "apache", "www-data", or something else."_
